crates/cext/src/transpiler/target.rs

```diff
@@ -869,6 +872,474 @@ pub unsafe extern "C" fn qk_target_num_instructions(target: *const Target) -> us
+///     qk_target_entry_add_property(entry, qargs, 2, 0.0, 0.1);
+///
+///     qk_target_add_instruction(target, entry);
+///
+///     bool has_0_1 = qk_target_op_has_qargs(target, 0, qargs, 2); // will be true
+///     bool has_0_2 = qk_target_op_has_qargs(target, 0, (uint32_t *){0, 2}, 2); // will be false
+///     bool has_global = qk_target_op_has_qargs(target, 0, NULL, 0); // will be false
+/// ```
+///
+/// # Safety
+///
+/// Behavior is undefined if ``QkTarget`` is not a valid, non-null pointer to a ``QkTarget``.
+#[unsafe(no_mangle)]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_op_has_qargs(
+    target: *const Target,
+    inst_idx: usize,
+    qargs: *const u32,
+    num_qubits: u32,
+) -> bool {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target_borrowed = unsafe { const_ptr_as_ref(target) };
+    // SAFETY: Per the documentation the qubits pointer is an array of num_qubits
+    // elements or a NULL pointer.
+    let parsed = unsafe { parse_qargs(qargs, num_qubits) };
+    target_borrowed
+        .get_from_index(inst_idx)
+        .map(|(_, inst_map)| inst_map.contains_key(&parsed))
+        .unwrap_or(false)
+}
+
+/// @ingroup QkTarget
+/// Retrieve the index in which some qargs are stored. Returns invalid number
+/// if not found.
+///
+/// @param target A pointer to the ``QkTarget``.
+/// @param index The index in which the gate is stored.
+/// @param qargs A pointer to the array of ``uint32_t`` qubit indices to
+///     check for, can be a null pointer to check for global properties.
+/// @param num_qubits The length of the qargs array.
+///
+/// # Example
+/// ```c
+///     QkTarget *target = qk_target_new(5);
+///
+///     QkTargetEntry *entry = qk_target_entry_new(QkGate_CX);
+///     uint32_t qargs[2] = {0, 1};
+///     qk_target_entry_add_property(entry, qargs, 2, 0.0, 0.1);
+///     qk_target_add_instruction(target, entry);
+///
+///     size_t idx_0_1 = qk_target_op_qargs_index(target, 0, qargs, 2);
+/// ```
+///
+/// # Safety
+///
+/// Behavior is undefined if ``QkTarget`` is not a valid, non-null pointer to a ``QkTarget``.
+#[unsafe(no_mangle)]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_op_qargs_index(
+    target: *const Target,
+    inst_idx: usize,
+    qargs: *const u32,
+    num_qubits: u32,
+) -> usize {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target_borrowed = unsafe { const_ptr_as_ref(target) };
+    // SAFETY: Per the documentation the qubits pointer is an array of num_qubits
+    // elements or a NULL pointer.
+    let parsed = unsafe { parse_qargs(qargs, num_qubits) };
+    if let Some(idx) = target_borrowed
+        .get_from_index(inst_idx)
+        .and_then(|(_, inst_map)| inst_map.get_index_of(&parsed))
+    {
+        idx
+    } else {
+        usize::MAX
+    }
+}
+
+/// @ingroup QkTarget
+/// Retrieve the qargs for the operation stored in its respective indices.
```

Comment:
The phrasing of this is a little odd to me. Maybe:

```suggestion
/// Retrieve the qargs for the operation by index.
```

Reply:
Added in ba679bd61eaa0ec7e3f103224407db1f3a497f89